components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -203,8 +203,30 @@ module radiation
    integer :: cldfsnow_idx = 0
 
    !needed for SPA
-   integer :: aer_tau_bnd_lw_mon_idx, aer_tau_bnd_sw_mon_idx,&
-              aer_ssa_bnd_sw_mon_idx, aer_asm_bnd_sw_mon_idx 
+   integer :: aer_tau_bnd_lw_idx(nlwbands), aer_tau_bnd_sw_idx(nswbands),&
+              aer_ssa_bnd_sw_idx(nswbands), aer_asm_bnd_sw_idx(nswbands)
+   !array of SPA pbuf indices
+   character(len=16), parameter :: aer_asm_sw_names(14) = &
```

Comment:
We can use these arrays in read_spa_data.F90, instead of specifying these fields again there. Use parameter variables instead of numeric literals.